tests/XML/CustomSigned.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimpleSAML\XMLSecurity\Test\XML;
+
+use DOMElement;
+use SimpleSAML\Assert\Assert;
+use SimpleSAML\XML\Chunk;
+use SimpleSAML\XML\Exception\MissingElementException;
+use SimpleSAML\XML\Exception\TooManyElementsException;
+use SimpleSAML\XML\Utils as XMLUtils;
+use SimpleSAML\XMLSecurity\XML\ds\Signature;
+use SimpleSAML\XMLSecurity\XML\AbstractSignedXMLElement;
+use SimpleSAML\XMLSecurity\XML\SignedElementInterface;
+use SimpleSAML\XMLSecurity\XML\SignedElementTrait;
+
+/**
+ * @package simplesamlphp\saml2
+ */
+final class CustomSigned extends AbstractSignedXMLElement
+{
+    use SignedElementTrait;
+
+
+    /**
+     * Create a class from XML
+     *
+     * @param \DOMElement $xml
+     * @return self
+     */
+    public static function fromXML(DOMElement $xml): object
```

Comment:
I feel like this method could be implemented generically. We don't really need to check the name of the element, since the _Signed*_ convention is just representing a signed element, but not an actual element named like that. That check will be done later when `fromXML()` is called on the actual class (e.g. `Assertion`). So then we just need to create the `Signed*` object by calling the constructor, passing the signature and the DOM. Something like this:

```php
    public static function fromXML(DOMElement $xml): SignedElementInterface
    {
        $signature = Signature::getChildrenOfClass($xml);
        Assert::minCount($signature, 1, MissingElementException::class);
        Assert::maxCount($signature, 1, TooManyElementsException::class);

        return new self($xml, $signature);
    }
```

If we make the `validate()` method return the `SignableElementInterface` object, then we don't even need to create that object first. We can just postpone the creation of the object until validation, which is also safer, I think.